internal/util/update/update_test.go

```diff
@@ -703,6 +704,144 @@ func TestCommand_Run_failInvalidPath(t *testing.T) {
 	}
 }
 
+func TestCommand_Run_badUpstreamLockRepo(t *testing.T) {
+	repos, w, clean := testutil.SetupReposAndWorkspace(t, map[string][]testutil.Content{
```

Comment:
In all these tests I think we should use the `TestSetupManager`. This makes sure we run the tests on a fork that has been properly cloned from an upstream repo.

When we want to make changes to the local fork after the initial fetch, we should use the `LocalChanges` property on the `TestSetupManager`. Just remember that this specified a complete new state for the local fork. Creating the invalid Kptfile configurations should be pretty straightforward with the pkgbuilder.

If the tests start to look very similar, consider using table-driven tests. There are examples in this file.

Reply:
It didn't seem like `LocalChanges` could be meaningfully changed after `TestSetupManager.Init()` is called, which was necessary to give `WithUpstreamLock()` a ref to the directory of the remote repo as well as a commit (and not just a commit index). I ended up using two instances of `TestSetupManager` to be able to get a remote repo path to give to the second instance which is actually being tested.